elide-core/src/main/java/com/yahoo/elide/core/QueryLogger.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2020, Yahoo Inc.
+ * Licensed under the Apache License, Version 2.0
+ * See LICENSE file in project root for terms.
+ */
+package com.yahoo.elide.core;
+
+import java.security.Principal;
+import java.util.Map;
+import java.util.Optional;
+import java.util.UUID;
+import javax.ws.rs.core.MultivaluedMap;
+
+/**
+ * Query Logger Interface for Elide
+ */
+public interface QueryLogger {
+
+    /**
+     * Accepts the incoming GraphQL query and notes the start time for the query
+     * @param queryId The RequestScope requestId.
+     * @param user The Principal user
+     * @param headers Http Request Headers
+     * @param apiVer API Version
+     * @param apiQuery QueryString of the requested URL request
+     */
+    void acceptQuery(UUID queryId, Principal user, Map<String, String> headers,
+                     String apiVer, String apiQuery);
+
+    /**
+     * Accepts the incoming JSON API query and notes the start time for the query
+     * @param queryId The RequestScope requestId.
+     * @param user The Principal user
+     * @param headers Http Request Headers
+     * @param apiVer API Version
+     * @param queryParams QueryParams for the incoming JSON API query
+     * @param path The apiQuery endpoint path for the incoming query
+     */
+    void acceptQuery(UUID queryId, Principal user, Map<String, String> headers, String apiVer,
+                     Optional<MultivaluedMap<String, String>> queryParams, String path);
+
+    /**
+     * Processes and logs all the queries from QueryDetail
+     * @param queryId The RequestScope requestId.
+     * @param qd The QueryDetail Object
+     */
+    void processQuery(UUID queryId, QueryDetail qd);
+
+    /**
+     * Cancels all queries currently running for a particular requestId
+     * @param queryId The RequestScope requestId.
+     */
+    void cancelQuery(UUID queryId);
```

Comment:
I don't think this is ever invoked.

Reply:
This is a member function of the QueryLogger class as indicated in the Carbon docs. Yep, you are right, this is never invoked.

Reply:
So I think we should call this function from transaction.cancel.  In order for that to work, we probably need to change the contract of the DataStoreTransaction.cancel to include the RequestScope.